File: Lib/Telerik/Examples/QuickStart/Docking/Programming/ProgrammableToolWnd.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms;
using Telerik.WinControls.UI.Docking;
using System.Drawing;

namespace Telerik.Examples.WinControls.Docking.Programming
{
	class ProgrammableToolWnd : DockWindow
	{
        private GroupBox groupBox;
		private Button buttonUnpin;
		private Button buttonPin;
		private Button buttonFloat;
		private Button buttonClose;
		private Button buttonDock;

        private Dictionary<string, Button> actionButtons = new Dictionary<string, Button>();

		

		public ProgrammableToolWnd()
		{
			InitializeComponent();
            //this.Activated += new EventHandler(ProgrammableToolWnd_Activated);

            actionButtons.Add("Unpin", buttonUnpin);
            actionButtons.Add("Pin", buttonPin);
            actionButtons.Add("Float", buttonFloat);
            actionButtons.Add("Close", buttonClose);
            actionButtons.Add("Dock", buttonDock);
		}

        void ProgrammableToolWnd_Activated(object sender, EventArgs e)
        {
            //if (this.DockingSite != null && this.DockingSite.DockingSiteType == DockingSiteType.FloatingContainerSite)
            //{
            //    this.buttonFloat.Enabled = false;
            //    this.buttonDock.Enabled = true;
            //    this.buttonPin.Enabled = false;
            //    this.buttonUnpin.Enabled = false;
            //}
        }

		private void InitializeComponent()
		{
			this.groupBox = new System.Windows.Forms.GroupBox();
			this.buttonClose = new System.Windows.Forms.Button();
			this.buttonUnpin = new System.Windows.Forms.Button();
			this.buttonPin = new System.Windows.Forms.Button();
			this.buttonFloat = new System.Windows.Forms.Button();
			this.buttonDock = new System.Windows.Forms.Button();
			this.groupBox.SuspendLayout();
			this.SuspendLayout();
			// 
			// groupBox
			// 
			this.groupBox.Controls.Add(this.buttonClose);
			this.groupBox.Controls.Add(this.buttonUnpin);
			this.groupBox.Controls.Add(this.buttonPin);
			this.groupBox.Controls.Add(this.buttonFloat);
			this.groupBox.Controls.Add(this.buttonDock);
			this.groupBox.Location = new System.Drawing.Point(18, 16);
			this.groupBox.Name = "groupBox";
			this.groupBox.Size = new System.Drawing.Size(140, 170);
			this.groupBox.TabIndex = 0;
			this.groupBox.TabStop = false;
			this.groupBox.Text = "Select an action:";
			// 
			// buttonClose
			// 
			this.buttonClose.Location = new System.Drawing.Point(30, 141);
			this.buttonClose.Name = "buttonClose";
			this.buttonClose.Size = new System.Drawing.Size(75, 23);
			this.buttonClose.TabIndex = 4;
			this.buttonClose.Text = "Close";
			this.buttonClose.UseVisualStyleBackColor = true;
			this.buttonClose.Click += new System.EventHandler(this.buttonClose_Click);
			// 
			// buttonUnpin
			// 
			this.buttonUnpin.Location = new System.Drawing.Point(30, 111);
			this.buttonUnpin.Name = "buttonUnpin";
			this.buttonUnpin.Size = new System.Drawing.Size(75, 23);
			this.buttonUnpin.TabIndex = 3;
			this.buttonUnpin.Text = "Unpin";
			this.buttonUnpin.UseVisualStyleBackColor = true;
			this.buttonUnpin.Click += new System.EventHandler(this.buttonUnpin_Click);
			// 
			// buttonPin
			// 
			this.buttonPin.Enabled = false;
			this.buttonPin.Location = new System.Drawing.Point(30, 81);
			this.buttonPin.Name = "buttonPin";
			this.buttonPin.Size = new System.Drawing.Size(75, 23);
			this.buttonPin.TabIndex = 2;
			this.buttonPin.Text = "Pin";
			this.buttonPin.UseVisualStyleBackColor = true;
			this.buttonPin.Click += new System.EventHandler(this.buttonPin_Click);
			// 
			// buttonFloat
			// 
			this.buttonFloat.Location = new System.Drawing.Point(30, 51);
			this.buttonFloat.Name = "buttonFloat";
			this.buttonFloat.Size = new System.Drawing.Size(75, 23);
			this.buttonFloat.TabIndex = 1;
			this.buttonFloat.Text = "Float";
			this.buttonFloat.UseVisualStyleBackColor = true;
			this.buttonFloat.Click += new System.EventHandler(this.buttonFloat_Click);
			// 
			// buttonDock
			// 
			this.buttonDock.Enabled = false;
			this.buttonDock.Location = new System.Drawing.Point(30, 21);
			this.buttonDock.Name = "buttonDock";
			this.buttonDock.Size = new System.Drawing.Size(75, 23);
			this.buttonDock.TabIndex = 0;
			this.buttonDock.Text = "Dock";
			this.buttonDock.UseVisualStyleBackColor = true;
			this.buttonDock.Click += new System.EventHandler(this.buttonDock_Click);
			// 
			// ProgrammableToolWnd
			// 
			this.BackColor = System.Drawing.Color.White;
			this.Controls.Add(this.groupBox);
			this.Location = new System.Drawing.Point(0, 26);
			this.Name = "ProgrammableToolWnd";
			this.Size = new System.Drawing.Size(184, 201);
			this.groupBox.ResumeLayout(false);
			this.ResumeLayout(false);

		}

		private void buttonDock_Click(object sender, EventArgs e)
		{
			this.buttonFloat.Enabled = true;
			this.buttonDock.Enabled = false;
			this.buttonUnpin.Enabled = true;
		}

        private void buttonFloat_Click(object sender, EventArgs e)
        {
            this.buttonFloat.Enabled = false;
            this.buttonDock.Enabled = true;
            this.buttonPin.Enabled = false;
            this.buttonUnpin.Enabled = false;

            Rectangle rect = new Rectangle(Control.MousePosition, this.Size);
        }

		private void buttonPin_Click(object sender, EventArgs e)
		{
			this.buttonPin.Enabled = false;
			this.buttonUnpin.Enabled = true;
		}

		private void buttonUnpin_Click(object sender, EventArgs e)
		{
			this.buttonPin.Enabled = true;
			this.buttonUnpin.Enabled = false;
            this.buttonFloat.Enabled = false;
		}

		private void buttonClose_Click(object sender, EventArgs e)
		{

		}

        public Button this[string buttonName]
        {
            get { return actionButtons[buttonName]; }
        }
	}
}

File: Lib/Telerik/Examples/QuickStart/Docking/Programming/ProgrammableToolWnd.vb
Imports System.Text
Imports Telerik.WinControls.UI.Docking

Namespace Telerik.Examples.WinControls.Docking.Programming
	Friend Class ProgrammableToolWnd
		Inherits DockWindow
		Private groupBox As GroupBox
		Private WithEvents buttonUnpin As Button
		Private WithEvents buttonPin As Button
		Private WithEvents buttonFloat As Button
		Private WithEvents buttonClose As Button
		Private WithEvents buttonDock As Button

		Private actionButtons As New Dictionary(Of String, Button)()



		Public Sub New()
			InitializeComponent()
			'this.Activated += new EventHandler(ProgrammableToolWnd_Activated);

			actionButtons.Add("Unpin", buttonUnpin)
			actionButtons.Add("Pin", buttonPin)
			actionButtons.Add("Float", buttonFloat)
			actionButtons.Add("Close", buttonClose)
			actionButtons.Add("Dock", buttonDock)
		End Sub

		Private Sub ProgrammableToolWnd_Activated(ByVal sender As Object, ByVal e As EventArgs)
			'if (this.DockingSite != null && this.DockingSite.DockingSiteType == DockingSiteType.FloatingContainerSite)
			'{
			'    this.buttonFloat.Enabled = false;
			'    this.buttonDock.Enabled = true;
			'    this.buttonPin.Enabled = false;
			'    this.buttonUnpin.Enabled = false;
			'}
		End Sub

		Private Sub InitializeComponent()
			Me.groupBox = New GroupBox()
			Me.buttonClose = New Button()
			Me.buttonUnpin = New Button()
			Me.buttonPin = New Button()
			Me.buttonFloat = New Button()
			Me.buttonDock = New Button()
			Me.groupBox.SuspendLayout()
			Me.SuspendLayout()
			' 
			' groupBox
			' 
			Me.groupBox.Controls.Add(Me.buttonClose)
			Me.groupBox.Controls.Add(Me.buttonUnpin)
			Me.groupBox.Controls.Add(Me.buttonPin)
			Me.groupBox.Controls.Add(Me.buttonFloat)
			Me.groupBox.Controls.Add(Me.buttonDock)
			Me.groupBox.Location = New Point(18, 16)
			Me.groupBox.Name = "groupBox"
			Me.groupBox.Size = New Size(140, 170)
			Me.groupBox.TabIndex = 0
			Me.groupBox.TabStop = False
			Me.groupBox.Text = "Select an action:"
			' 
			' buttonClose
			' 
			Me.buttonClose.Location = New Point(30, 141)
			Me.buttonClose.Name = "buttonClose"
			Me.buttonClose.Size = New Size(75, 23)
			Me.buttonClose.TabIndex = 4
			Me.buttonClose.Text = "Close"
			Me.buttonClose.UseVisualStyleBackColor = True
'			Me.buttonClose.Click += New System.EventHandler(Me.buttonClose_Click)
			' 
			' buttonUnpin
			' 
			Me.buttonUnpin.Location = New Point(30, 111)
			Me.buttonUnpin.Name = "buttonUnpin"
			Me.buttonUnpin.Size = New Size(75, 23)
			Me.buttonUnpin.TabIndex = 3
			Me.buttonUnpin.Text = "Unpin"
			Me.buttonUnpin.UseVisualStyleBackColor = True
'			Me.buttonUnpin.Click += New System.EventHandler(Me.buttonUnpin_Click)
			' 
			' buttonPin
			' 
			Me.buttonPin.Enabled = False
			Me.buttonPin.Location = New Point(30, 81)
			Me.buttonPin.Name = "buttonPin"
			Me.buttonPin.Size = New Size(75, 23)
			Me.buttonPin.TabIndex = 2
			Me.buttonPin.Text = "Pin"
			Me.buttonPin.UseVisualStyleBackColor = True
'			Me.buttonPin.Click += New System.EventHandler(Me.buttonPin_Click)
			' 
			' buttonFloat
			' 
			Me.buttonFloat.Location = New Point(30, 51)
			Me.buttonFloat.Name = "buttonFloat"
			Me.buttonFloat.Size = New Size(75, 23)
			Me.buttonFloat.TabIndex = 1
			Me.buttonFloat.Text = "Float"
			Me.buttonFloat.UseVisualStyleBackColor = True
'			Me.buttonFloat.Click += New System.EventHandler(Me.buttonFloat_Click)
			' 
			' buttonDock
			' 
			Me.buttonDock.Enabled = False
			Me.buttonDock.Location = New Point(30, 21)
			Me.buttonDock.Name = "buttonDock"
			Me.buttonDock.Size = New Size(75, 23)
			Me.buttonDock.TabIndex = 0
			Me.buttonDock.Text = "Dock"
			Me.buttonDock.UseVisualStyleBackColor = True
'			Me.buttonDock.Click += New System.EventHandler(Me.buttonDock_Click)
			' 
			' ProgrammableToolWnd
			' 
			Me.BackColor = Color.White
			Me.Controls.Add(Me.groupBox)
			Me.Location = New Point(0, 26)
			Me.Name = "ProgrammableToolWnd"
			Me.Size = New Size(184, 201)
			Me.groupBox.ResumeLayout(False)
			Me.ResumeLayout(False)

		End Sub

		Private Sub buttonDock_Click(ByVal sender As Object, ByVal e As EventArgs) Handles buttonDock.Click
			Me.buttonFloat.Enabled = True
			Me.buttonDock.Enabled = False
			Me.buttonUnpin.Enabled = True
		End Sub

		Private Sub buttonFloat_Click(ByVal sender As Object, ByVal e As EventArgs) Handles buttonFloat.Click
			Me.buttonFloat.Enabled = False
			Me.buttonDock.Enabled = True
			Me.buttonPin.Enabled = False
			Me.buttonUnpin.Enabled = False

			Dim rect As New Rectangle(Control.MousePosition, Me.Size)
		End Sub

		Private Sub buttonPin_Click(ByVal sender As Object, ByVal e As EventArgs) Handles buttonPin.Click
			Me.buttonPin.Enabled = False
			Me.buttonUnpin.Enabled = True
		End Sub

		Private Sub buttonUnpin_Click(ByVal sender As Object, ByVal e As EventArgs) Handles buttonUnpin.Click
			Me.buttonPin.Enabled = True
			Me.buttonUnpin.Enabled = False
			Me.buttonFloat.Enabled = False
		End Sub

		Private Sub buttonClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles buttonClose.Click

		End Sub

		Default Public ReadOnly Property Item(ByVal buttonName As String) As Button
			Get
				Return actionButtons(buttonName)
			End Get
		End Property
	End Class
End Namespace
